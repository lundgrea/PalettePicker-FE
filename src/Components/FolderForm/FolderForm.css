.FolderForm {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  height: 150px;
  background-color: red;
}

#addFolder-input {
  width: 75%;
  margin: auto;
  border-radius: 5px;
  height: 35px;
  font-size: 15px;
}

#addFolder-button {
  width: 50%;
  margin: auto;
  border-radius: 5px;
  height: 25px;
}