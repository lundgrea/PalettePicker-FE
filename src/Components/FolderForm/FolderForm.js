import React, { Component } from 'react'
import './FolderForm.css'

class FolderForm extends Component {
  constructor() {
    super();
    this.state = {
      folderName: "",
      error: ""
    };
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value});
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const newFolder = {
      id: Date.now(),
      name: this.state.name,
      palettes: []
    }
    this.addFolder(newFolder);
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({folderName: ""})
  }

  render() {
    return (
      <form className="FolderForm">
        <h2 id="addFolder-text">Create a New Folder</h2>
        <input
        id="addFolder-input"
        type="text"
        placeholder="Folder Name"
        name="folderName"
        value={this.state.folderName}
        onChange={this.handleChange}
        required/>
        <button 
        id="addFolder-button"
        >Add Folder</button>
      </form>
    )
  }
}

export default FolderForm;