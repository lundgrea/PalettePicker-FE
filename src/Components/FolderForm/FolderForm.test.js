import React from "react";
import { shallow } from "enzyme";
import FolderForm from "./FolderForm";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });


let folders = {folders: [{name: "Cruising", id: 99}, {name: "Just Joshin", id: 33}]}

describe("FolderForm", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<FolderForm folders={folders} addNewFolder={jest.fn()} getAllFolders={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  });


  it("should begin with an empty state", () => {
    const wrapper = shallow(<FolderForm folders={folders} addNewFolder={jest.fn()} getAllFolders={jest.fn()}/>);
    expect(wrapper.state("folderName")).toEqual("")
    expect(wrapper.state("error")).toEqual("")
  });

  it("should call handleChange method when change happens", () => {
    const wrapper = shallow(<FolderForm folders={folders} addNewFolder={jest.fn()} getAllFolders={jest.fn()}/>);
    const mockEvent = { target: { name: "folderName" , value: "The Finisher"}};
    wrapper.find("input").simulate("change", mockEvent);
    expect(wrapper.state("folderName")).toEqual("The Finisher");
  });


  it('should fire handleSubmit on the click of the Add Folder Button', () => {
    const wrapper = shallow(<FolderForm folders={folders} addNewFolder={jest.fn()} getAllFolders={jest.fn()}/>);
    const mockEvent = { target: { name: "folderName" , value: "The Finisher"}};
    wrapper.find("input").simulate("change", mockEvent);
    expect(wrapper.state("folderName")).toEqual("The Finisher");
    const mockClickEvent = { preventDefault: jest.fn() };
    wrapper.find("img").simulate("click", mockClickEvent);
    wrapper.instance().checkForDuplicates(mockClickEvent, "The Finisher")
    expect(wrapper.state("folderName")).toEqual("")
  })

  it("should reset state to an empty string", () => {
    const wrapper = shallow(<FolderForm folders={folders} addNewFolder={jest.fn()} getAllFolders={jest.fn()}/>);
    const mockEvent = { target: { name: "folderName" , value: "The Finisher"}};
    wrapper.find("input").simulate("change", mockEvent);
    expect(wrapper.state("folderName")).toEqual("The Finisher");
    const mockClickEvent = { preventDefault: jest.fn() };
    wrapper.find("img").simulate("click", mockClickEvent);
    expect(wrapper.state("folderName")).toEqual("");
  });
});
