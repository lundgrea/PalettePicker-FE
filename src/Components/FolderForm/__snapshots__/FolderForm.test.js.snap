// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderForm should match the snapshot 1`] = `
<form
  className="FolderForm"
>
  <label
    id="addFolder-text"
  >
    Create a New Folder
  </label>
  <div
    id="input-button-container"
  >
    <input
      id="addFolder-input"
      name="folderName"
      onChange={[Function]}
      placeholder="Enter New Folder Name"
      required={true}
      type="text"
      value=""
    />
    <img
      alt="addFolder-button"
      id="addFolder-button"
      onClick={[Function]}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "add.svg",
        }
      }
    />
  </div>
</form>
`;
